openpype/hosts/maya/api/lib.py

```diff
@@ -3176,38 +3178,78 @@ def _colormanage(**kwargs):
 def parent_nodes(nodes, parent=None):
     # type: (list, str) -> list
     """Context manager to un-parent provided nodes and return them back."""
-    import pymel.core as pm  # noqa
 
-    parent_node = None
-    delete_parent = False
+    def _as_mdagpath(node):
+        """Return MDagPath for node path."""
+        if not node:
+            return
+        sel = OpenMaya.MSelectionList()
+        sel.add(node)
+        return sel.getDagPath(0)
 
+    # We can only parent dag nodes so we ensure input contains only dag nodes
+    nodes = cmds.ls(nodes, type="dagNode", long=True)
+    if not nodes:
+        # opt-out early
+        yield
+        return
+
+    parent_node_path = None
+    delete_parent = False
     if parent:
         if not cmds.objExists(parent):
-            parent_node = pm.createNode("transform", n=parent, ss=False)
+            parent_node = cmds.createNode("transform",
+                                          name=parent,
+                                          skipSelect=False)
             delete_parent = True
         else:
-            parent_node = pm.PyNode(parent)
+            parent_node = parent
+        parent_node_path = cmds.ls(parent_node, long=True)[0]
```

Comment:
If parent_node = parent, then could it be the same:

```suggestion
        parent_node_path = cmds.ls(parent, long=True)[0]
```

Reply:
Parent can be created [here](https://github.com/ynput/OpenPype/pull/4724/files#diff-0f94b40dc0b348d7194c61cb01a57668eb3490d22cdbff89d37dd552287fccadR3201) and in that case `parent_node` != `parent`.